src/Component/RuleGenerator/RuleGeneratorWindow.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+import { Rnd } from 'react-rnd';
+
+import {
+  Rule
+} from 'geostyler-style';
+
+import { Data } from 'geostyler-data';
+
+import './RuleGeneratorWindow.css';
+import { Button } from 'antd';
+
+import { localize } from '../LocaleWrapper/LocaleWrapper';
+import en_US from '../../locale/en_US';
+import RuleGenerator from './RuleGenerator';
+
+const _isEqual = require('lodash/isEqual');
+// i18n
+export interface RuleGeneratorWindowLocale {
+  ruleGenerator: string;
+}
+
+// default props
+export interface RuleGeneratorWindowDefaultProps {
+  locale: RuleGeneratorWindowLocale;
+}
+
+// non default props
+export interface RuleGeneratorWindowProps extends Partial<RuleGeneratorWindowDefaultProps> {
+  rules?: Rule[];
+  internalDataDef?: Data;
+  x?: number;
+  y?: number;
+  onClose?: () => void;
+  onRulesChange?: (rules: Rule[]) => void;
+}
+
+/**
+ * Symbolizer editorwindow UI.
+ */
+export class RuleGeneratorWindow extends React.Component<RuleGeneratorWindowProps> {
+
+  public static defaultProps: RuleGeneratorWindowDefaultProps = {
+    locale: en_US.GsRuleGeneratorWindow
+  };
+
+  public shouldComponentUpdate(nextProps: RuleGeneratorWindowProps): boolean {
+    return !_isEqual(this.props, nextProps);
+  }
+
+  static componentName: string = 'SymbolizerEditorWindow';
+
+  render() {
+    const {
+      x,
+      y,
+      onClose,
+      rules,
+      onRulesChange,
+      locale,
+      internalDataDef
+    } = this.props;
+
+    return (
+      ReactDOM.createPortal(
+        <Rnd
```

Comment:
Please add `bounds` property, so that the window cannot be dragged outside of the browser window. See https://github.com/bokuweb/react-rnd#bounds-string